scripts/configurator/parser/cmake_var.py

```diff
@@ -1,9 +1,108 @@
-from typing import Dict, Any, Type, Mapping
+from typing import Dict, Any, Optional, Type, Mapping, List
+import re
+from sys import stderr
+
+
+class metadata:
+   keyword_expr = re.compile(r"([A-Z_]+)")
```

Comment:
I didn't think we need to use a regex for the keywords. In the simple if/elif impl. I would have used `if line.startswith("GROUP ")` etc. But, with this implementation this is understandable. Just make it be: `^([A-Z_]+)` so that's more explicit we're requiring that at the beginning of the line.

Reply:
Note: I added a comment below about this regex. It needs more changes than simply `^` at the beginning.

Reply:
Ok, i will refer to the comment below